server/game/cards/locations/05/oldwyk.js

```diff
@@ -7,30 +7,42 @@ class OldWyk extends DrawCard {
         this.reaction({
             when: {
                 onChallenge: (event, challenge) =>
-                    challenge.attackingPlayer === this.controller
-                    && challenge.challengeType === 'power'
+                    challenge.attackingPlayer === this.controller &&
+                    challenge.challengeType === 'power' &&
+                    this.anyDrownedGodInDeadPile()
             },
             cost: ability.costs.kneelSelf(),
             handler: () => {
-                var card = _.last(this.controller.deadPile.filter(c => c.hasTrait('Drowned God')));
-
-                if(!card) {
-                    this.game.addMessage('{0} finds no character in their dead pile to put into play with {1}', this.controller, this);
-                    return;
-                }
+                let card = _.last(this.controller.deadPile.filter(c => c.hasTrait('Drowned God')));
 
                 this.controller.putIntoPlay(card);
                 this.game.currentChallenge.addAttacker(card);
 
-                this.game.addMessage('{0} uses {1} to put into play {2} as an attacker from their dead pile', this.controller, this, card);
+                this.game.addMessage('{0} kneels {1} to put {2} into play from their dead pile as an attacker', 
+                                      this.controller, this, card);
 
-                this.untilEndOfChallenge(ability => ({
-                    match: card,
-                    effect: ability.effects.returnToHandOrDeckBottom()
-                }));
+                this.game.once('afterChallenge:interrupt', (event, challenge) => this.resolveAfterChallenge(challenge, card));
```

Comment:
> 2. This is purely hypothetical, but during the D of DUCK the following abilities trigger; what is the proper order?

The Red Wedding-Lord of the Crossing-Old Wyk-Spearmaiden-Superior Claim.

> 2) 
Interrupts - The Red Wedding
Delayed effects - Lord of the Crossing / Old Wyk / Spearmaiden (first player decides)
Reactions: Superior Claim

I used afterChallenge:interrupt to mirror Spearmaiden/Dolorous Edd, but if afterChallenge:cancelinterrupt is correct I'll change them all.